VitruvioHost/Plugins/Vitruvio/Source/VitruvioEditor/Private/VitruvioEditorModule.cpp

```diff
@@ -124,6 +124,32 @@ FLevelEditorModule::FLevelViewportMenuExtender_SelectedActors LevelViewportConte
 
 } // namespace
 
+void VitruvioEditorModule::PostUndoRedo()
+{
+	// We want to call PostUndoRedo on the VitruvioComponent after the undo action has completed (note that the overriden PreEditUndo/PostEditUndo on
+	// the Component itself is called during the undo operation and always before its owners Actor undo/redo has completed). We also need to check if
+	// the VitruvioComponent was involved in the undo/redo action.
+	const FTransaction* LastTransaction = GEditor->Trans->GetTransaction(GEditor->Trans->GetQueueLength() - 1);
```

Comment:
This entire function setup looks really hacky to me. If we really need to do call PostUndoRedo later again could we not simply schedule the undo/redo call from within the component (e.g. by adding it to a list, setting a flag ect..) and then calling postUndoRedo on the marked components?